python/runtime/pai/submitter.py

```diff
@@ -0,0 +1,379 @@
+
+    with open(path.join(cwd, "requirements.txt"), "w") as require:
+        require.write(TF_REQUIREMENT)
+    copy_python_package("sqlflow_submitter", cwd)
+    copy_python_package("sqlflow_models", cwd)
+    copy_custom_package(estimator, cwd)
+
+    os.execl("tar", "czf", JOB_ARCHIVE_FILE, "./sqlflow_submitter",
+             "./sqlflow_models", "requirements.txt", TRAIN_PARAMS_FILE)
+
+
+def save_model_to_sqlfs(datasource, model_oss_path, model_name):
+    # (TODO: save model to sqlfs)
+    pass
+
+
+# (TODO: lhw) adapt this interface after we do feature derivation in Python
+def submit_pytf_train(datasource, estimator, select, validation_select,
+                      model_params, model_name, pre_trained_model,
+                      **train_params):
+    """This function submit PY-TF train task to PAI platform
+
+    Args:
+        datasource: string
+            Like: odps://access_id:access_key@service.com/api?curr_project=test_ci&scheme=http
+        estimator: string
+            The name of tensorflow estimator
+        select: string
+            The SQL statement for selecting data for train
+        validation_select: string
+            Ths SQL statement for selecting data for validation
+        model_params: dict
+            Params for training, crossponding to WITH clause
+        pre_trained_model: string
+            The pre-trained model name to load
+        train_params: dict
+            Extra train params, they will be passed to sqlflow_submitter.tensorflow.train
+    """
+
+    # prepare params for tensorflow train,
+    # the params will be pickled into train_params.pkl
+    params = locals()
+    del params["train_params"]
+    params.update(train_params)
+    params["entry_type"] = "train"
+
+    cwd = os.getcwd()
+    conf = cluster_conf.get_cluster_config(model_params)
+
+    train_table, val_table = create_train_and_eval_tmp_table(
+        select, validation_select, datasource)
+    params["pai_table"], params["pai_val_table"] = train_table, val_table
+
+    # clean target dir
+    path_to_save = get_oss_model_save_path(datasource, model_name)
+    path_to_load = get_oss_model_save_path(datasource, pre_trained_model)
+    project = get_project(datasource)
+
+    if path_to_load == "" or path_to_load != path_to_save:
+        clean_oss_model_path(path_to_save + "/")
+
+    # zip all required resource to a tarball
+    prepare_archive(cwd, conf, project, estimator, model_name, train_table,
+                    val_table, path_to_save, params)
+
+    # submit pai task to execute the training
+    cmd = get_pai_tf_cmd(conf, JOB_ARCHIVE_FILE, PARAMS_FILE,
+                         ENTRY_DIR + "tensorflow.py", model_name, path_to_save,
+                         train_table, val_table, "", project, cwd)
+    submit_pai_task(cmd, datasource)
+
+    # save trained model to sqlfs
+    save_model_to_sqlfs(datasource, path_to_save, model_name)
+    drop_tmp_tables([train_table, val_table], datasource)
```

Comment:
We should call `drop_tmp_tables` when there's any exception or error occurs in above process.